tests/plugins/imagebase/features/_helpers/tools.js

```diff
@@ -0,0 +1,209 @@
+/* global pasteFiles */
+
+( function() {
+	'use strict';
+
+	function objToArray( obj ) {
+		var tools = CKEDITOR.tools;
+
+		return tools.array.map( tools.objectKeys( obj ), function( key ) {
+			return obj[ key ];
+		} );
+	}
+
+	window.imageBaseFeaturesTools = {
+		/*
+		 * Main assertion for pasting files.
+		 *
+		 * @param {CKEDITOR.editor} editor
+		 * @param {Object} options
+		 * @param {File[]} [options.files=[]] Files to be dropped.
+		 * @param {Function} options.callback Function to be called after the paste event.
+		 * Params:
+		 *
+		 * * `CKEDITOR.plugins.widget[]` widgets - Array of widgets in a given editor.
+		 * * `CKEDITOR.eventInfo` evt - Paste event.
+		 * * `CKEDITOR.eventInfo/undefined` uploadEvt - Upload event, available ony if `options.fullLoad` was set
+		 *	to `true` and at least one widget was found.
+		 * @param {String} [options.dataValue=null] HTML data to be put into the clipboard if any.
+		 * @param {Boolean} [options.fullLoad=false] If `true` assertion will wait for `uploadDone` of `uploadFailed`
+		 * event of the first widget.
+		 */
+		assertPasteFiles: function( editor, options ) {
+			var files = options.files || [],
+				callback = options.callback;
+
+			editor.once( 'paste', function( evt ) {
+				// Unfortunately at the time being we need to do additional timeout here, as
+				// the paste event gets cancelled.
+				setTimeout( function() {
+					var widgets = objToArray( editor.widgets.instances ),
+						listeners = [];
+
+					function wrappedCallback( uploadEvt ) {
+						// In case we listen for `upload*` events only first event should be handled (to not cause multiple)
+						// resume calls.
+						if ( listeners.length ) {
+							listeners = CKEDITOR.tools.array.filter( listeners, function( curListener ) {
+								curListener.removeListener();
+								return false;
+							} );
+						}
+
+						resume( function() {
+							callback( objToArray( editor.widgets.instances ), evt, uploadEvt );
+						} );
+					}
+
+					if ( options.fullLoad && widgets.length ) {
+						for ( var i = widgets.length - 1; i >= 0; i-- ) {
+							listeners.push( widgets[ i ].once( 'uploadDone', wrappedCallback, null, null, 999 ) );
+							listeners.push( widgets[ i ].once( 'uploadFailed', wrappedCallback, null, null, 999 ) );
+						}
+					} else {
+						wrappedCallback();
+					}
+				}, 0 );
+			}, null, null, -1 );
+
+			// pasteFiles is defined in clipboard plugin helper.
+			pasteFiles( editor, files, options.dataValue );
+
+			wait();
+		},
+
+		progress: {
+			/*
+			 * Returns a type that implements a progress indicator that puts a white overflow
+			 * on a image, and removes it indicating the progress.
+			 *
+			 * @param {CKEDITOR.editor} editor
+			 * @param {Function} ProgressReporter Base type for progress indicator
+			 * {@link CKEDITOR.plugins.imageBase.progressReporter}.
+			 * @returns {Function}
+			 */
+			getProgressOverlapType: function( editor, ProgressReporter ) {
```

Comment:
Well we don't have an official and strict rule about this. My intention was to uppercase a local variable in case it stores a type. Does it make sense?